tracer/src/Datadog.Trace/Telemetry/Transports/JsonTelemetryTransport.cs

```diff
@@ -0,0 +1,94 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Serialization;
+
+namespace Datadog.Trace.Telemetry.Transports;
+
+internal class JsonTelemetryTransport : ITelemetryTransport
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<JsonTelemetryTransport>();
+    internal static readonly JsonSerializerSettings SerializerSettings = new()
+    {
+        NullValueHandling = NullValueHandling.Ignore,
+        ContractResolver = new DefaultContractResolver { NamingStrategy = new SnakeCaseNamingStrategy(), }
+    };
+
+    private readonly IApiRequestFactory _requestFactory;
+    private readonly Uri _endpoint;
+
+    public JsonTelemetryTransport(IApiRequestFactory requestFactory, Uri baseEndpoint)
+    {
+        _requestFactory = requestFactory;
+        _endpoint = new Uri(baseEndpoint, TelemetryConstants.TelemetryPath);
+    }
+
+    public async Task<TelemetryPushResult> PushTelemetry(TelemetryData data)
+    {
+        try
+        {
+            // have to buffer in memory so we know the content length
+            var serializedData = SerializeTelemetry(data);
+            var bytes = Encoding.UTF8.GetBytes(serializedData);
+
+            var request = _requestFactory.Create(_endpoint);
+
+            request.AddHeader(TelemetryConstants.ApiVersionHeader, data.ApiVersion);
+            request.AddHeader(TelemetryConstants.RequestTypeHeader, data.RequestType);
+
+            var response = await request.PostAsync(new ArraySegment<byte>(bytes), "application/json").ConfigureAwait(false);
+            if (response.StatusCode is >= 200 and < 300)
+            {
+                Log.Debug("Telemetry sent successfully");
+                return TelemetryPushResult.Success;
+            }
+            else if (response.StatusCode == 404)
+            {
+                Log.Debug("Error sending telemetry: 404. Disabling further telemetry, as endpoint not found");
+                return TelemetryPushResult.FatalError;
+            }
+            else
+            {
+                Log.Debug<int>("Error sending telemetry {StatusCode}", response.StatusCode);
+                return TelemetryPushResult.TransientFailure;
+            }
+        }
+        catch (Exception ex) when (IsFatalException(ex))
+        {
+            Log.Warning(ex, "Error sending telemetry data, unable to communicate with endpoint");
+            return TelemetryPushResult.FatalError;
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "Error sending telemetry data");
+            return TelemetryPushResult.TransientFailure;
+        }
+    }
+
+    // Internal for testin
```

Comment:
```suggestion
    // Internal for testing
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2617/commits/6f413f5d269f7fce697910ea20b080f9eaaea659